src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1465,6 +1467,75 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Retrieve good values for extended persistence, and separate the diagrams into the ordinary, relative, extended+ and extended- subdiagrams. Need extend_filtration to be called first!
+   * @param[in] dgm Persistence diagram obtained after calling this->extend_filtration and this->get_persistence.
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the second one is Relative, the third one is Extended+ and the fourth one is Extended-.
+   */
+  std::vector<std::vector<std::pair<int, std::pair<double, double>>>> convert(const std::vector<std::pair<int, std::pair<double, double>>>& dgm){
```

Comment:
The name `convert` seems awfully generic.

Reply:
Agreed. I changed to `compute_extended_persistence_subdiagrams`